pyproject.toml

```diff
@@ -31,7 +31,7 @@ generate-setup-file = true
 
 [tool.poetry.dependencies]
 python = "^3.11"
-poetry = "^1.3.2"
+poetry = "^1.1.14"
```

Comment:
This poetry version is too old to work with dependency groups (see below in the file).

But poetry shouldn’t be a dependency of the main package in the first place (no idea what came over me there to add it here) – would you move it to `tool.poetry.group.dev.dependencies` and rebuild the lock file?

Reply:
Will this solve the RPM issue as well?

Reply:
Yep, the installed Python package shouldn’t depend on Poetry at all (regardless of its version).

Reply:
For some reason, the version stepdown does help solve the problem as mentioned in https://github.com/fedora-infra/mirrors-countme/pull/103#issuecomment-1687490147 while the v1.3.2 which was there previously does not.

I suppose that is because we mentioned Poetry as a project runtime dependency and not as a development dependency like you mentioned, @nphilipp. The generated tarball file should have a good old `setup.py` file to work with and should not require Poetry at all.

Reply:
On second thought - does this project really rely on Poetry as a "development dependency"? A bunch of projects that I have worked on do not even need its mention in either of the runtime dependencies section or the development dependencies section and they still work fine.

Reply:
Yes it does, kinda. Of course there’s the boot-strapping problem: Poetry needs to be installed before you can use it to evaluate `pyproject.toml` and install the specified dependencies in the first place.

It’s not pointless though:

- Listing it as a dev dependency lets us document the minimum poetry version
- It ensures that developers and CI alike use the same (pinned) version, so depenency resolution gets the same results everywhere. Having the same version can reduce churn in the `poetry.lock` file, e.g. when two people use different poetry versions which format things differently. This can be quite a formatting/sorting ping pong if e.g. dependabot/Renovate use one version and I or another developer uses another. 😉

Reply:
For the record, the Poetry people recommend not installing Poetry in the same venv that it's managing for the application. If you run `poetry install --no-dev` (or `--without-group dev` now), Poetry used to remove the development dependencies, and thus remove itself. Apparently this has changed since 1.2.0, but before that I've been hit by it in monorepos such as datanommer where you'd install all the subrepos in the same venv.
Anyway, this may not be an issue anymore :-)